src/core_modules/capture-core/components/Pages/common/EnrollmentOverviewDomain/EnrollmentPageLayout/hooks/useWidgetColumns.js

```diff
@@ -0,0 +1,90 @@
+// @flow
+import React, { useCallback, useMemo } from 'react';
+import log from 'loglevel';
+import { errorCreator } from '../../../../../../../capture-core-utils';
+
+import { WidgetTypes } from '../DefaultEnrollmentLayout.constants';
+import type { ColumnConfig, PageLayoutConfig, WidgetConfig } from '../DefaultEnrollmentLayout.types';
+
+type Props = {
+    pageLayout: PageLayoutConfig,
+    availableWidgets: $ReadOnly<{ [key: string]: WidgetConfig }>,
+    props: Object,
+};
+
+const MemoizedWidgets: { [key: string]: React$ComponentType<any> } = {};
+const UnsupportedWidgets: { [key: string]: boolean } = {};
+
+const renderWidget = (widget: ColumnConfig, availableWidgets, props) => {
+    const { type } = widget;
+
+    if (type.toLowerCase() === WidgetTypes.COMPONENT) {
```

Comment:
Are we going to support other types of widgets in the future?

Reply:
Thanks for the review! Yes, we will extend this config to also support Enrollment Dashboard plugins. 😊
https://dhis2.atlassian.net/browse/DHIS2-16305